src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -58,10 +57,22 @@ public static VisualStudioInstance GetInstance(string workingDirectory)
                 discoveryType: DiscoveryType.DotNetSdk);
         }
 
-        private static string GetDotNetBasePath(string workingDirectory)
+        public static IEnumerable<VisualStudioInstance> GetInstances(string workingDirectory)
+        {
+            var basePaths = GetDotNetBasePaths(workingDirectory);
+
+            foreach (var basePath in basePaths)
+            {
+                yield return GetInstance(workingDirectory, basePath);
```

Comment:
nit: It would be cleaner to filter out `null` instances here and revert the extra check in `QueryVisualStudioInstances`.

Reply:
Fixed